### PR TITLE
hardening share type detection, return "not shared" if type is unknown

### PR DESCRIPTION
**Pull request by AndyScherzinger**
fix for https://github.com/nextcloud/android/issues/1536

Prevent ```null``` for (yet) unknown or unsupported share types. 

@schiessle do you know if there are more share types then:
*  0 - Shared by user
*  1 - Shared by group
*  3 - Shared by public link
*  4 - Shared by e-mail
*  5 - Shared by contact
*  6 - Shared by federation

### CODE REVIEW

File: src/com/owncloud/android/lib/resources/shares/ShareType.java [hunk 1]
```diff
@@ -75,7 +76,8 @@ public static ShareType fromValue(int value)
             return CONTACT;
         case 6:
             return FEDERATED;
+        default:
+            return NO_SHARED;
         }
-        return null;
     }
 };
```
Comment (mario) [inline src/com/owncloud/android/lib/resources/shares/ShareType.java]: New line please.

Comment (AndyScherzinger) [reply src/com/owncloud/android/lib/resources/shares/ShareType.java]: instead of the removed line?

Comment (mario) [reply src/com/owncloud/android/lib/resources/shares/ShareType.java]: No, after };

**Comment by mario** [comment]
👍

File: src/com/owncloud/android/lib/resources/shares/ShareType.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/com/owncloud/android/lib/resources/shares/ShareType.java]: Done via https://github.com/nextcloud/android-library/pull/89/commits/b0ebaae55bc3cea88bfa15ce0081b285c2d9f0e3

**Comment by mario** [comment]
Saw, already approved.

**Comment by mario** [comment]
@tobiasKaminsky can we get an approve? ;)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can we get an approve? This is a fix for app version 2.0.0

**Comment by tobiasKaminsky** [comment]
:+1:

**Comment by tobiasKaminsky** [comment]
@mario please re-approve  :-)

**Comment by mario** [comment]
👍

**Pull request merged**